hyundai_kia_connect_api/HyundaiApiIN.py

```diff
@@ -0,0 +1,968 @@
+        vehicle.air_control_is_on = get_child_value(state, "airCtrlOn")
+        vehicle.smart_key_battery_warning_is_on = get_child_value(
+            state, "smartKeyBatteryWarning"
+        )
+
+        vehicle.data = state
+
+    def _get_time_from_string(self, value, timesection) -> dt.datetime.time:
+        if value is not None:
+            lastTwo = int(value[-2:])
+            if lastTwo > 60:
+                value = int(value) + 40
+            if int(value) > 1260:
+                value = dt.datetime.strptime(str(value), "%H%M").time()
+            else:
+                d = dt.datetime.strptime(value, "%I%M")
+                if timesection > 0:
+                    d += dt.timedelta(hours=12)
+                value = d.time()
+        return value
+
+    def get_vehicles(self, token: Token) -> list[Vehicle]:
+        url = self.SPA_API_URL + "vehicles"
+        response = requests.get(
+            url, headers=self._get_authenticated_headers(token)
+        ).json()
+        _LOGGER.debug(f"{DOMAIN} - Get Vehicles Response: {response}")
+        _check_response_for_errors(response)
+        result = []
+        for entry in response["resMsg"]["vehicles"]:
+            entry_engine_type = None
+            if entry["type"] == "GN":
+                entry_engine_type = ENGINE_TYPES.ICE
+            elif entry["type"] == "EV":
+                entry_engine_type = ENGINE_TYPES.EV
+            elif entry["type"] == "PHEV":
+                entry_engine_type = ENGINE_TYPES.PHEV
+            elif entry["type"] == "HV":
+                entry_engine_type = ENGINE_TYPES.HEV
+            vehicle: Vehicle = Vehicle(
+                id=entry["vehicleId"],
+                name=entry["nickname"],
+                model=entry["vehicleName"],
+                registration_date=entry["regDate"],
+                VIN=entry["vin"],
+                timezone=self.data_timezone,
+                engine_type=entry_engine_type,
+            )
+            result.append(vehicle)
+        return result
+
+    def get_last_updated_at(self, value) -> dt.datetime:
+        _LOGGER.debug(f"{DOMAIN} - last_updated_at - before {value}")
+        if value is None:
+            value = dt.datetime(2000, 1, 1, tzinfo=self.data_timezone)
+        else:
+            m = re.match(r"(\d{4})(\d{2})(\d{2})(\d{2})(\d{2})(\d{2})", value)
+            value = dt.datetime(
+                year=int(m.group(1)),
+                month=int(m.group(2)),
+                day=int(m.group(3)),
+                hour=int(m.group(4)),
+                minute=int(m.group(5)),
+                second=int(m.group(6)),
+                tzinfo=self.data_timezone,
+            )
+
+        _LOGGER.debug(f"{DOMAIN} - last_updated_at - after {value}")
+        return value
+
+    def force_refresh_vehicle_state(self, token: Token, vehicle: Vehicle) -> None:
+        # FIXME: Gowtham - No force refresh
+        _LOGGER.error(f"Disabling force refresh")
```

Comment:
Add domain like other logging to clarify what is throwing this.    I assume this exists since force refresh command doesn't work yet?